### PR TITLE
Minq draft 21

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: handshake-layer.go [hunk 1]
```diff
@@ -173,10 +173,12 @@ func (h *HandshakeLayer) sendAlert(err Alert) error {
 
 func (h *HandshakeLayer) ReadMessage() (*HandshakeMessage, error) {
 	var hdr, body []byte
-	err := WouldBlock
+	err := error(nil)
```
Comment (marten-seemann) [inline handshake-layer.go]: Better: `var err error`

File: conn.go [hunk 2]
```diff
@@ -728,3 +729,33 @@ func (c *Conn) SendKeyUpdate(requestUpdate bool) error {
...
+	}
+
+	h0 := c.state.cryptoParams.Hash.New().Sum(nil)
+	tmpSecret := deriveSecret(c.state.cryptoParams, c.state.exporterSecret, label, h0)
+
+	hc := c.state.cryptoParams.Hash.New().Sum(context)
+	return HkdfExpandLabel(c.state.cryptoParams.Hash, tmpSecret, "exporter", hc, keyLength), nil
+}
+
+func (c *Conn) GetConnectionState() ConnectionState {
```
Comment (bifurcation) [inline conn.go]: Nit: Think `State()` would be more aesthetic

File: conn.go [hunk 3]
```diff
@@ -173,9 +173,10 @@ var (
 )
 
 type ConnectionState struct {
-	HandshakeComplete bool                // TLS handshake is complete
-	CipherSuite       CipherSuite         // cipher suite in use (TLS_RSA_WITH_RC4_128_SHA, ...)
-	PeerCertificates  []*x509.Certificate // certificate chain presented by remote peer
+	HandshakeState   string              // string representation of the handshake state.
+	CipherSuite      CipherSuiteParams   // cipher suite in use (TLS_RSA_WITH_RC4_128_SHA, ...)
+	PeerCertificates []*x509.Certificate // certificate chain presented by remote peer TODO(ekr@rtfm.com): implement
+	NextProto        string              // Selected ALPN proto
```
Comment (bifurcation) [inline conn.go]: This is starting to look mighty similar to [`ConnectionParams`](https://github.com/bifurcation/mint/blob/master/state-machine.go#L70).  Is there some way to de-dup?

File: frame-reader.go [hunk 4]
```diff
@@ -2,8 +2,7 @@
 // This is used for both TLS Records and TLS Handshake Messages
 package mint
 
-import (
-)
+import ()
```
Comment (bifurcation) [inline frame-reader.go]: Nit: You can just delete this line.

File: conn.go (hunk 3 above)
Comment (ekr) [inline conn.go]: Probably but I think too hard for this PR. See #136

**Pull request merged**